File: leetcode_75/Queue/python/933.py
from collections import deque


class RecentCounter:

    def __init__(self):
        self.queue = deque()

    def ping(self, t: int) -> int:
        self.queue.append(t)
        left = t - 3000
        while self.queue[0] < left:
            self.queue.popleft()

        return len(self.queue)

File: leetcode_75/Queue/cpp/933.cpp
#include <iostream>
#include <queue>

using namespace std;

class RecentCounter {
public:
    RecentCounter() {} 

    int ping(int t) {
        this->counter.push(t);
        int left = t - 3000;
        while (this->counter.front() < left)
        {
            this->counter.pop();
        }

        return this->counter.size();
    }

private:
    queue<int> counter;
};

int main() {
    RecentCounter rc;
    rc.ping(10); // Добавляет элемент 10 в очередь rc.counter
    return 0;
}